halo2_proofs/src/poly/kzg/commitment.rs

```diff
@@ -132,6 +132,27 @@ where
         }
     }
 
+    /// Initializes parameters for the curve through existing parameters
+    /// k, n, g, g_lagrange, g2, s_g2
+    pub fn from_params(
+        &self,
+        k: u32,
+        n: u64,
+        g: Vec<E::G1Affine>,
+        g_lagrange: Vec<E::G1Affine>,
```

Comment:
This could be an option. 

User then can do a tradeoff between space vs. performance.

If Option.is_some(_) => Use the internal `g_lagrange`.
Else, compute `g_lagrange` based on `g`.